pkg/virt-operator/resource/generate/components/prometheus.go

```diff
@@ -98,7 +105,9 @@ func NewPrometheusRuleSpec(ns string, workloadUpdatesEnabled bool) *v1.Prometheu
 							"runbook_url": runbookUrlBasePath + "VirtAPIDown",
 						},
 						Labels: map[string]string{
-							"severity": "critical",
+							severityAlertLabelKey:  "critical",
```

Comment:
Instead of manually adding the label to each rule, how about iterating all rules and adding the label? Something along the lines of:

```go
	for _, group := range ruleSpec.Groups {
		for _, rule := range group.Rules {
			if rule.Labels == nil {
				rule.Labels = make(map[string]string)
			}
			rule.Labels[virtv1.AppPartOfLabel] = productName
			rule.Labels[virtv1.AppComponentLabel] = productComponent
		}
	}
```

I can see that some of the rules don't have a `severity` label attached, and for them you didn't add the new `part-of` and `component` labels. Was that intentional? I think we can still add the labels to all of them (the above code snippets does that).

Reply:
@zcahana All alerts rules include the new labels from what I saw. Adding the labels automatically could be a good , Can we add it only for the "Alerts" rules and not all the rules that also include recording rules?

Reply:
Right, looked again and all rules without labels are recording rules (I failed to spot that before).

And yeah, we can definitely restrict that when iterating the rules (this way we ensure we don't miss any):

```go
	for _, group := range ruleSpec.Groups {
		for _, rule := range group.Rules {
                        if rule.Alert == "" {
                                continue
                        }
			if rule.Labels == nil {
				rule.Labels = make(map[string]string)
			}
			rule.Labels[virtv1.AppPartOfLabel] = productName
			rule.Labels[virtv1.AppComponentLabel] = productComponent
		}
	}
```

Reply:
/hold